src/plugins/intel_gpu/src/graph/graph_optimizer/mark_runtime_skippable_nodes.cpp

```diff
@@ -95,5 +96,22 @@ void mark_runtime_skippable_nodes::run(program& p) {
             node.can_be_optimized(true);
             GPU_DEBUG_TRACE_DETAIL << "[mark_runtime_skippable_nodes] : " << node.id() << " can_be_optimized" << std::endl;
         });
+        program_helpers::do_for_types<broadcast>(*node, [](broadcast_node& node){
+            auto impl_params = node.get_kernel_impl_params();
+            if (node.is_output()
+                || node.has_fused_primitives()
+                || (impl_params->get_input_layout(0).format != impl_params->get_output_layout().format)
+                || (impl_params->get_input_layout(0).data_type != impl_params->get_output_layout().data_type))
+                return;
+
+            if (node.is_dynamic()) {
+                // If the user is reorder, it could be fused to broadcast in the remove_redundant_reorders pass.
+                // In this case, broadcast can not be optimized due to different input and output shapes.
+                if (node.have_user_with_type<reorder>() && node.get_users().size() == 1)
+                    return;
+                node.can_be_optimized(true);
```

Comment:
Is there any more possibility to reduce the target scope? I am afraid this will let most of the broadcasts be marked as can_be_opt so that less chance of the memory reuse, though the ratio of the actual optimizable cases is considered as small.. For example could we exclude more of "apparently not optimizable cases" like [-1, 1, -1, -1] [-1, 16, -1, -1]

Reply:
Oh... I'll try to reduce the target scope as you reviewed. Thank you.

Reply:
I added an additional condition not to skip broadcast. Thank you.